File: java/src/Diamond.java
public class Diamond {
    interface IA { default void foo() { System.out.println("IA"); } }
    interface IB extends IA { default void foo() { System.out.println("IB"); } }
    interface IC extends IA { default void foo() { System.out.println("IC"); } }

    static class D implements IB, IC {
        @Override
        public void foo() { // Must override to resolve ambiguity
            IB.super.foo(); // Explicitly choose IB's implementation
        }
    }

    public static void main(String[] args) {
        D d = new D();
        d.foo(); // Output: IB
    }
}

File: kotlin/src/diamond.kt
interface IA { fun foo() { println("IA") } }
interface IB : IA { override fun foo() { println("IB") } }
interface IC : IA { override fun foo() { println("IC") } }

class D : IB, IC {
    override fun foo() {
        super<IB>.foo() // Explicitly choose IB's implementation
    }
}

fun main() {
    D().foo() // Output: IB
}